src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -157,7 +157,8 @@ public async Task Invoke(HttpContext context)
             IDictionary<string, string> commands = this.requestParser.ParseRequestCommands(context);
             if (commands.Count > 0)
             {
-                foreach (string command in commands.Keys)
+                List<string> commandKeys = new List<string>(commands.Keys);
```

Comment:
I would have used `ToArray()` here but a list will do for now.

Reply:
I dread to use System.Linq in active/hot path 😉